localstack/utils/cloudformation/template_deployer.py

```diff
@@ -1048,6 +1048,10 @@ def deploy_resource(resource_id, resources, stack_name):
 
 
 def delete_resource(resource_id, resources, stack_name):
+    for res_id, res in resources.items():
+        if res['ResourceType'] == 'AWS::S3::Bucket':
+            S3_NOTIFICATIONS.pop(res['PhysicalResourceId'], None)
```

Comment:
Instead of importing `S3_NOTIFICATIONS` here, can we better add a new function `remove_bucket_notification(..)` to `s3_listener.py`? The dict `S3_NOTIFICATIONS` should remain an implementation detail that is only locally accessible the file it is contained in (for better encapsulation).